### DISCUSSION TITLE
Camera Stops Working....  Solved!

### DISCUSSION THREAD

**Discussion by RuthlessTX** [Guilouz/Creality-Helper-Script-Wiki#531] [Issues]
Camera suddenly stop working in Fluidd

Settings > Camera > Click your camera and change the following values

Camera Url Stream: /webcam?action=stream
Camera Url Snapshot: /webcam?action=snapshot

Use the full URI structure!
Please be sure to replace the ip address with the IP of your printer!

Camera Url Stream: http://XXX.XXX.XXX.XXX:4409/webcam/?action=stream
Camera Url Snapshot: http://XXX.XXX.XXX.XXX:4409/webcam/?action=snapshot

Credit to:  morriscode (https://www.reddit.com/user/morriscode/)

**Comment by jattie-ire** [comment]
Interesting, thanks for sharing.

**Comment by jattie-ire** [comment]
This unfortunately did not resolve my issue and I think that is because the issue manifests in different ways. In my case the cam_app service terminates and will not restart unless I reboot the system.

**Comment by NexGen-3D-Printing** [comment]
My Camera went down on the K1C aswell, reboot fixed, on generic machines I just restart Crowsnest, not sure what we can do on the K1 series?

**Comment by mfraser** [comment]
Mine doesn't stop working, it just starts to flicker after a while. Can I just use 127.0.0.1 as the IP address in case the IP address of the printer changes?

**Reply by jattie-ire** [reply]
I think it should work.

Flickering is indicative of a service restarting, so I would be curious to see your system logs. Can you run dmesg on the terminal and share the output when you observe the flickering?

The services on the K1 are not very easy to trace but what I managed to figure out so far is that the AI service seems to be like a master service that then shares the stream with mjpeg- streamer. 

I tried to get mjpeg-streamer to stream the camera directly but the formats are not standard webcam, and information on the camera is quite sparse, so I have not been able to get it working directly yet.

**Reply by mfraser** [reply]
127.0.0.1 doesn't work I had to enter the IP address of the printer.

I think it might be better described as looking like the image isn't always fully loading, like when you only get the top part of a JPEG.

**Comment by RuthlessTX** [comment]
Localhost might also work

Joel M. Gabel

"Any sufficiently advanced technology is indistinguishable from magic." -
Arthur C. Clarke

**Reply by mfraser** [reply]
I don't think it will as it seems to use that URL in the page, so it would point to localhost on the computer where the page is open.
When I start to get the flickering of the webcam is also when the previews stop appearing on the display and I get a white square instead.

**Comment by RuthlessTX** [comment]
Mine is working great with the IP for over a week now.  Have 2 rooted K1Max
printers running that config.

Joel M. Gabel

"Any sufficiently advanced technology is indistinguishable from magic." -
Arthur C. Clarke

**Comment by NexGen-3D-Printing** [comment]
Looks like I missed the camera restart macro, I initially disable all from the web GUI as there was way to many, so all good for me, mine still occasionaly goes down but the restart macro fixes it up.

Still cant get the camera to put out a descent frame rate though.

**Comment by RuthlessTX** [comment]
Yeah the framerate is garbage on both of mine (Seems to be a newer lens or
camera on my 12/23 production date vs my 2/23 printer)

On Fri, May 17, 2024 at 3:31 AM NexGen-3D ***@***.***> wrote:

> Looks like I missed the camera restart macro, I initially disable all from
> the web GUI as there was way to many, so all good for me, mine still
> occasionaly goes down but the restart macro fixes it up.
>
> Still cant get the camera to put out a descent frame rate though.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Guilouz/Creality-Helper-Script-Wiki/discussions/531#discussioncomment-9467560>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BHT66CX3GOFNYP72DK23NOTZCW545AVCNFSM6AAAAABGY4ASTGVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TINRXGU3DA>
> .
> You are receiving this because you authored the thread.Message ID:
> <Guilouz/Creality-Helper-Script-Wiki/repo-discussions/531/comments/9467560
> @github.com>
>


-- 
Joel M. Gabel

"Any sufficiently advanced technology is indistinguishable from magic." -
Arthur C. Clarke

**Comment by TheWebMachine** [comment]
Have you looked at this discussion here? #508

I've been working on the camera issues for months now with little success. We should join forces! haha 

I had to stop testing for now because I found a defect in my K1C heater core (the mount flanges sheer off *way* too easily) and am awaiting warranty replacement parts shipping from Creality.

**Comment by RuthlessTX** [comment]
Oh that’s a whole new set of issues.  Mine was completely fluidd based.

Joel M. Gabel

"Any sufficiently advanced technology is indistinguishable from magic." -
Arthur C. Clarke

**Reply by TheWebMachine** [reply]
Well, so far in my testing, issues like white/blank preview, low framerates, etc have been tracing back to the image processing SoC in these cameras overheating. They run very hot and this gets obviously worse if the printer is enclosed.

I suppose I'm popping in here to see if any of the issues you are having might be unknowingly related to thermal problems with the camera. I initially thought my issues were entirely software/fluidd/moonraker/mainsail related, too, even fussing with URIs and IPs thinking it was helping...until I cracked open the camera and put a heatsink on the SoC inside. While it didn't solve all my issues, the cam stopped cutting out during a long print job 95% of the time and my framerates are almost always stable at 15fps.

You said you have two camera modules between your two manufacturer dates? I'm curious which variants they are. I'm counting no less than 4 so far, because Creality keeps tweaking these mid-production. The problem is, all these variables are making it nearly impossible for us to keep track of the camera issues (among other things).

**Reply by jattie-ire** [reply]
I managed to get to the bottom of my issues and it manifested in a completely different way. 

I saw these laser messages in the device manager log and I noticed their timing coincides consistently with the USB disconnect message in the system dmesg logs. 

==> /usr/data/creality/userdata/log/device_manager.log <==
[2024/05/05-17:14:07:241303]-[INFO]-[device_manager.c](494) laser_set_state: ubus_notify: {"laser_plugged":0}
[2024/05/05-17:14:07:777128]-[INFO]-[device_manager.c](494) laser_set_state: ubus_notify: {"laser_plugged":1}

==> dmesg_2.log <==
2024-05-05T17:14:07,326819+01:00 usb 1-1.1: USB disconnect, device number 35
2024-05-05T17:14:07,533828+01:00 usb 1-1.1: new full-speed USB device number 36 using dwc2
2024-05-05T17:14:07,633578+01:00 usb 1-1.1: New USB device found, idVendor=1a86, idProduct=55d3
2024-05-05T17:14:07,641104+01:00 usb 1-1.1: New USB device strings: Mfr=0, Product=2, SerialNumber=3
2024-05-05T17:14:07,649475+01:00 usb 1-1.1: Product: USB Single Serial
2024-05-05T17:14:07,654369+01:00 usb 1-1.1: SerialNumber: 5719024095
2024-05-05T17:14:07,660514+01:00 cdc_acm 1-1.1:1.0: ttyACM0: USB ACM device

As an experiment I disconnected the lidar sensor and the problem went away, like in no more camera failure. 

So it seems the lidar connecting disconnecting eventually results in the camera_app terminating and resulting in the camera stream going dead and the only recovery from that is a complete system reboot. 

I am getting a new LIDAR under warranty.

**Reply by TheWebMachine** [reply]
Keep an eye on it and report back, for sure. In my thermal tests, I was getting USB disconnects in dmesg too, but I have the K1C with no lidar at all. It may be an issue, in general, with the USB controller or kernel module for it, if both lidar and camera were experiencing issues at the same time. While I can't speak for the OG K1, I'm pretty sure the K1C and K1 Max are using the same mainboard.

**Reply by jattie-ire** [reply]
I think the camera_app controls both lidar and webcams and if you look at the bottom of the lidar unit there is actually another camera in there too. The mjpeg-streamer doesnt get a direct camara feed, it looks at a file system provided by camera_app, so if you kill camera_app you should in theory be able to run mjpeg-steamer independently without the need of camera_app, but then if you interrogate the camera with v4l2, you dont get a normal usb camera and this is where I got stuck, I cannot figure out how to get mjpeg-streamer to run the video stream directly without the intermediate camera_app. 

Maybe you found something I could not figure out so far, if so please share your insights and let's dump it in a wiki.

**Reply by AcBCDN** [reply]
Wow! That worked.  The camera was failing < 3 mins in. Now I am over 1 hr in and it is still working. Thanks for the efforts.

**Reply by jattie-ire** [reply]
So another case of LIDAR malfunction manifesting camera failure?

**Comment by RuthlessTX** [comment]
Interesting as the first thing I did was remove the lidar assembly from
both my K1Max’s…. Good catch on your diagnosis!

Joel M. Gabel

"Any sufficiently advanced technology is indistinguishable from magic." -
Arthur C. Clarke

**Reply by jattie-ire** [reply]
Took me a few weeks of diligent slog work. The flavour of Linux on the K1 is quite strange and non standard, but I managed to get a decent way into figuring it our, but the camera part is still a bit of a mystery.

**Comment by NexGen-3D-Printing** [comment]
> Well, so far in my testing, issues like white/blank preview, low framerates, etc have been tracing back to the image processing SoC in these cameras overheating. They run very hot and this gets obviously worse if the printer is enclosed.
> 
> I suppose I'm popping in here to see if any of the issues you are having might be unknowingly related to thermal problems with the camera. I initially thought my issues were entirely software/fluidd/moonraker/mainsail related, too, even fussing with URIs and IPs thinking it was helping...until I cracked open the camera and put a heatsink on the SoC inside. While it didn't solve all my issues, the cam stopped cutting out during a long print job 95% of the time and my framerates are almost always stable at 15fps.
> 
> You said you have two camera modules between your two manufacturer dates? I'm curious which variants they are. I'm counting no less than 4 so far, because Creality keeps tweaking these mid-production. The problem is, all these variables are making it nearly impossible for us to keep track of the camera issues (among other things).

I think this is totaly my issue with K1C, I only print ABS, and I have it fully enclosed with almost no way for heat to escape, chamber will get over 60C after a few hours and the camera always craps itself in the middle of long prints, I will look at a thermal solution for it.

I have had many issues with cameras dying from heat, all of my nozzle cams die quite quickly, I have collection going, and a number of RPI cameras all dead, these all came out of my machines with active heated chambers.

Ill test mine and report back if I see any improvments.

**Reply by TheWebMachine** [reply]
Pro tip: the medium size heatsink from any generic Pi4 heatsink trio kit fits the SoC in the camera perfectly, but you'll need to cut a hole in rear camera cover for it to stick out (and get some airflow). I think I posted a pic of mine with the heatsink in the discussion I linked above.

**Reply by TheWebMachine** [reply]
Also, I still occasionally get a camera fail on super long prints, but the frequency has dropped substantially. Unfortunately, with the positioning in the enclosure, there is not an easy way to add more cooling to it beyond the basic heatsink and cover cut out without heavy modifications to the mount, the enclosure, or both.

**Reply by TheWebMachine** [reply]
(I started mocking up a new camera enclosure with 1/8" water cooling loop and then snapped back to reality. 🤣)

**Reply by jattie-ire** [reply]
> 
> I have had many issues with cameras dying from heat, all of my nozzle cams die quite quickly, I have collection going, and a number of RPI cameras all dead, these all came out of my machines with active heated chambers.
> 
> 
> 
I was not aware of a nozzle cam. It's this something you retrofitted or do I have one and just don't know about it?

**Reply by TheWebMachine** [reply]
> I was not aware of a nozzle cam. It's this something you retrofitted or do I have one and just don't know about it?

It's only on the K1 Max as part of the lidar assembly. Not the K1C, if you have that one.

**Reply by jattie-ire** [reply]
I have a K1 max too. So can I view the nozzle cam or is it used by lidar and AI features only?

**Comment by NexGen-3D-Printing** [comment]
Let me see if I can design something, I think the angle on the camera is also garbage, so I may as will make a new enclosure with some adjustment.

**Reply by TheWebMachine** [reply]
Yeah, something adjustable that might allow room for something like a little 20mm fan back there - like these that could maybe even be powered from the cam's own 5v line: https://a.co/d/iU9QUZR

I couldn't find any reliable source for 15mm fans, which would be a better fit; they just aren't commodity enough at 15mm to be readily available.

However, keep in mind the reach of the gantry. The cam is already SUPER close to the extruder at the far end of the X axis. Creality also screwed the pooch with clearances of the rear corner steppers cuz you can totally run into them with the extruder and shift the known position if you aren't careful with super large prints, but I digress.

My design skills begin to wane the smaller the parts, so you'll certainly come up with something way better than I could. haha

**Comment by NexGen-3D-Printing** [comment]
Im pretty clever with things like this, I'll use a 20x6 mini blower from GDStime, I use these on my custom JetStream cooler for Nema14 rounds.

**Reply by TheWebMachine** [reply]
Nice!

**Comment by NexGen-3D-Printing** [comment]
This should do the trick, Ill run the fan from auxillery blower output, as I have removed that from my machine and it now runs an active filter, this little blower can run at the same time as I print with the filter running at all times.

![Image01](https://github.com/Guilouz/Creality-Helper-Script-Wiki/assets/20825418/2f748386-48f2-465b-9e4f-e4d54e2258d6)
![Image02](https://github.com/Guilouz/Creality-Helper-Script-Wiki/assets/20825418/eb3a68b3-3537-4009-97d7-3edda746aaa2)

**Reply by jattie-ire** [reply]
Good job on the design, looks great. 

Two thoughts: 

- Raspberry pi fans are 5v. 
- How about moving the camera outside of the enclosure?

**Comment by TheWebMachine** [comment]
So, since the lidar is part of another camera assembly on the Max, perhaps it is that camera that is dying and killing the connection for the lidar, as well. These cameras have an SoC that is rather impressive for its size. dual-core MIPS 1.2Ghz, 1 or 2GB of DDR2 RAM, separate image co-processors, USB controller, etc. That's a lot for such a small package to not be cooled in any way. At idle, open chamber, not printing, just feeding video, *outside of its own enclosure,* the SoC in the chamber camera gets to about 60°C! In *its own enclosure,* it climbs even higher than that...then you start printing with a build plate sitting right underneath it, let alone enclosing it for things ABS.  I suspect the problem is exponentially worse on the hotend lidar/camera assembly.

![321401543-5d14623d-4725-49de-a487-9b2b91dc838e](https://github.com/Guilouz/Creality-Helper-Script-Wiki/assets/1045052/bbe44a64-5e8d-49da-a499-93391fc11f04)


And what happens when a CPU/APU/SoC gets too hot? PC Overclock enthusiasts will know that that is when things get glitchy. As best I can tell so far, once the SoC gets above a certain internal temp, everything starts glitching. Frame rates drop first, as the SoC attempts to down-clock, but this still isn't enough. So, it just hits a ceiling and eventually the USB controller glitches and resets (I think), killing the connection. 

Part of the issue may be that the software side of things doesn't handle this reset gracefully, preventing the cam[s and lidar] from recovering until reboot. 

So, yes, the handling of the issue is likely flawed in software (typical Creality), but the issue seems to begin with thermal overload glitching out the hardware first. It also explains why the lidar/camera assembly has needed replaced so many times for some Max users; repeated extreme thermal stress is going to eventually complete kill the SoC[s] permanently.

**Reply by TheWebMachine** [reply]
The "moral" of this story is, I think: 3D printing needs a new camera design methodology to account for the thermal shortcomings of most of the current "off-the-shelf" camera modules out there. Even standard CMOS-based camera image sensors are going to degrade from the heat over time, which surely accounts for some camera failures people experience.

**Comment by NexGen-3D-Printing** [comment]
Ive just ordered in a new 5v 2006 blower from GDSTime as I cant find the one I had kicking about, It will be a week or so before I can test it.

![IMG_20240519_074725_933](https://github.com/Guilouz/Creality-Helper-Script-Wiki/assets/20825418/a229b879-b4a4-4f15-b357-2850a28c6b71)
![IMG_20240519_074718_449](https://github.com/Guilouz/Creality-Helper-Script-Wiki/assets/20825418/fd7720c9-9e3c-467b-9a1c-be77c732da45)
![IMG_20240519_074655_934](https://github.com/Guilouz/Creality-Helper-Script-Wiki/assets/20825418/e8a201ff-6e12-43cb-9a74-1518db8cb43a)

**Reply by TheWebMachine** [reply]
Nice! 😯

**Comment by NexGen-3D-Printing** [comment]
Okay, sorry for the long wait, I had multiple nozzle blockage's which turned into a completely destroyed hotend, not a fan of the heater setup and the full length Nozzle, mostly pointless as it flows no better than a standard V6, I tried everything to eliminate nozzle seizing inside the heater, thermal paste, anti-seize, Copper based anti-seize and now I'm using high temp brake calliper slide grease, such a stupid design, well I digress.

Here is the camera mod, and it works, no more lockups, even when printing PC-CF:

**Printables:**[ K1C Camera Mod](https://www.printables.com/model/904768-nexgen3d-creality-k1c-camera-fix)

**YouTube:**[ K1C Camera Mod](https://www.youtube.com/watch?v=7YnPlfGt-qc)

**Reply by TheWebMachine** [reply]
Awesome! Thanks for the work on this. I'll get one printed and a fan ordered and see how it goes for me. 

I'm gonna see about powering the fan off the camera's 5v USB rail for a cleaner install. Since the fan is only rated 100mA, it shouldn't be an issue since the T31 chip is also rated at less than 100mA; the rest of the camera components surely aren't gonna pull another 300mA to overload the 0.5A of the USB2 port. My only concern is potential noise on the VCC, but I doubt it will be an issue. 🤞

In your experience with these fans, how's their lifespan? I'm hoping to find a source other than AliEx (I have the *worst* luck with them.), but if I can't I'm just curious how many I should order to have spares. 😎

**Reply by NexGen-3D-Printing** [reply]
Im running it off the camera power now, the fan is 5v, no issues so far, Camera hasn't stop at all, its been 100% stable, normally 15mins and its dead every print.

Second black wire in on the plug is ground, red is 5v.

GDSTime fans are awesome, its all I use now, I was using Sunon Maglev, but they dont make them in a wide range, the dual ball bearing ones from GDStime are 99% as good as Sunon Maglev and are rated for long term use.

**Comment by Supabad** [comment]
i installed creality helper recently and noticed this camera issues also , i only print with pla. and have one additional cam installed. its working but after a while it disconnects. the only thing to get them back is to run /etc/init.d/S50usb_camera restart on my k1.

10 seconds after that the cameras are back online.

**Reply by NexGen-3D-Printing** [reply]
Since installing the cooling fan, I have never seen the camera fail, not even once, it was definately heat related for me, I print ABS and Polycarbonate constantly in mine, so high chamber temps, my camera use to fail withing minutes every single time, even restarting the camera would result in it failing rapidly straight after.

**Comment by davidgarnerjr** [comment]
Ive made the Camera URL stream and snapshot changes, printed the camera fan mount with blower fan and installed a thermal heat break on the SOC along with unplugging the lidar. I even ordered a new K1 AI cam from Creality and nothing has fixed my camera, it still fails during prints. I just run a Wyze cam external now to check on it, extremely annoying but it works.

**Reply by NexGen-3D-Printing** [reply]
Maybe your control board is faulty? I believe there is additional hardware on there for the camera, both of my machines are perfect, well as perfect as we can get with such under-powered hardware and wifi connection, my K1 Max performs better than my K1C's as it has ethernet and a lower overall chamber temp.
